cloud/amazon/datapipeline.py

```diff
@@ -0,0 +1,470 @@
+    return pipeline_exists_timeout(client, dp_id, timeout=timeout)
+
+
+def delete_pipeline(client, module):
+    """Deletes pipeline
+
+    """
+    dp_name = module.params.get('name')
+    timeout = module.params.get('timeout')
+
+    try:
+        dp_id = pipeline_id(client, dp_name)
+        _delete_dp_with_check(dp_id, client, timeout)
+        changed = True
+    except DataPipelineNotFound:
+        changed = False
+    except TimeOutException:
+        module.fail_json(msg=('Data Pipeline {} failed to delete'
+                              'within timeout {} seconds').format(dp_name,
+                                                                  timeout))
+
+    result = {'id': dp_id,
+              'msg': 'Data Pipeline {} deleted'.format(dp_name)}
+
+    return (changed, result)
+
+
+def build_unique_id(dp_name, dp_version):
+    return '{0}-{1}'.format(dp_name, dp_version)
+
+
+def create_pipeline(client, module):
+    """Creates datapipeline. Uses version and uniqueId to achieve idempotency.
+
+    """
+    dp_name = module.params.get('name')
+    dp_version = module.params.get('version')
+    objects = module.params.get('objects', None)
+    description = module.params.get('description', '')
+    if not dp_version:
+        module.fail_json(msg='Version is required when creating the pipeline!')
+    if objects:
+        parameters = module.params.get('parameters')
+        values = module.params.get('values')
+
+    unique_id = build_unique_id(dp_name, dp_version)
+    result = dict()
+    create_dp = False
+    try:
+        dp_id = pipeline_id(client, dp_name)
+        dp_unique_id = pipeline_uniqueid(client, dp_id)
+        # delete existing pipeline before creating new one
+        if dp_unique_id != unique_id:
+            changed, _ = delete_pipeline(client, module)
+            create_dp = True
+        else:
+            changed = False
+            msg = 'Data Pipeline {} version {} is present'.format(dp_name,
+                                                                  dp_version)
+            result = {'id': dp_id,
+                      'unique_id': dp_unique_id,
+                      'msg': msg}
+    except DataPipelineNotFound:
+        create_dp = True
+
+    if create_dp:
+        dp = client.create_pipeline(name=dp_name,
+                                    uniqueId=unique_id,
+                                    description=description)
+        dp_id = dp['pipelineId']
+        if objects:
+            client.put_pipeline_definition(pipelineId=dp_id,
+                                           pipelineObjects=objects,
+                                           parameterObjects=parameters,
+                                           parameterValues=values)
+            changed = True
+
+        result = {'id': dp_id,
+                  'msg': 'Data Pipeline {} created'.format(dp_name)}
+
+    return (changed, result)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
```

Comment:
argument spec should include  ec2 arg spec.  

argument_spec = ec2_argument_spec()